ci-operator/step-registry/jboss-fuse/run-tests/jboss-fuse-run-tests-commands.sh

```diff
@@ -0,0 +1,220 @@
+        containers:
+          - name: xpaas-qe
+            ports:
+              - containerPort: 22
+                protocol: TCP
+              - containerPort: 80
+                protocol: TCP
+              - containerPort: 443
+                protocol: TCP
+            env:
+            - name: MVN_ARGS
+              value: -Dlpinterop
+            - name: MVN_SETTINGS_PATH
+              value: /tmp/custom.settings.xml
+            - name: MVN_PROFILES
+              value: test-fuse,openshift4-current
+            - name: TEST_EXPR
+              value: ${interopTestList[$i]}
+            - name: OPENSHIFT_NAMESPACE
+              value: ${namespace}
+            - name: NAMESPACE
+              value: ${namespace}
+            - name: NAMESPACE_PREFIX
+              value: ${namespace}
+            - name: SUREFIRE_REPORTS_FOLDER
+              value: /deployments/xpaas-qe/test-fuse/target/surefire-reports
+            securityContext:
+              privileged: true
+            volumeMounts:
+              - name: mvn-settings
+                mountPath: /tmp/
+              - name: test-properties
+                mountPath: /mnt/
+              - name: xpaas-qe-volume
+                mountPath: /tmp/xtf-oc-cache/
+                subPath: xtf-oc-cache
+              - name: xpaas-qe-volume
+                mountPath: /tmp/surefire-reports/
+                subPath: surefire-reports
+              - name: xpaas-qe-volume
+                mountPath: /tmp/log/
+                subPath: log
+              - name: xpaas-qe-volume
+                mountPath: /deployments/.m2/
+                subPath: maven-repo
+              - name: xpaas-qe-volume
+                mountPath: /tmp/reports
+                subPath: reports
+              - name: xpaas-qe-volume
+                mountPath: /tmp/tests
+                subPath: tests
+EOF
+    sleep 30
+done
+}
+
+function check_tests() {
+	mapfile -t podlist < <(oc get pods -n jboss-fuse-interop -l application=xpaas-qe --no-headers=true | awk '{print $1}')
+	echo "LIST ----"
+	echo "${podlist[@]}"
+
+  while (( ${#podlist[@]} )); do
+    echo "Number running pods still: ${#podlist[@]}"
+    for i in "${!podlist[@]}"; do
+       currentPod=${podlist[$i]}
+       echo "Evaluating $currentPod ..."
+       podlog=$(while read line; do echo $line; done <<< "$(oc logs --tail=50 $currentPod -n jboss-fuse-interop)")
+       if [[ "$podlog" == *"BUILD SUCCESS"* ]]; then
+         echo "pod $currentPod terminating"
+         if [[ "$podlog" == *"Failures: 0, Errors: 0"* ]]; then
+           echo "$currentPod was successful"
+           echo "get surefire xml pod reports"
+           oc -n jboss-fuse-interop exec $currentPod -- /bin/bash -c 'cp -rf /deployments/xpaas-qe/test-fuse/target/surefire-reports/*.xml /tmp/reports' || true
+           oc -n jboss-fuse-interop exec $currentPod -- /bin/bash -c 'cp -f /deployments/xpaas-qe/test-fuse/log/test.log /tmp/tests/'$currentPod'.log' || true
```

Comment:
@smongiar After we are getting `$currentPod was successful` how about copying logs directly from $currentPod to `ARTIFACTS DIR`. In this way we'll get the artifacts stored for each DC before getting deleted. And we won't have duplicated also. WDYT?
```suggestion
           oc cp jboss-fuse-interop/"${currentPod}":deployments/xpaas-qe/test-fuse/log/test.log "${ARTIFACT_DIR}/tests"
           oc cp jboss-fuse-interop/"${currentPod}":/deployments/xpaas-qe/test-fuse/target/surefire-reports/*.xml "${ARTIFACT_DIR}"
```

Reply:
it's ok, but the issue with duplicates is about failing pods :)

Reply:
exactly, actually, I'm introducing a variable that would stop the regeneration of pods in case of "max retries limit" is reached to copy logs anyway